tvtk/pyface/ui/qt4/scene.py

```diff
@@ -338,7 +339,7 @@ def __init__(self, parent=None, **traits):
         super(Scene, self).__init__(parent, **traits)
 
         # Setup the default picker.
-        self.picker = picker.Picker(self)
+        self.picker = picker.Picker(weakref.proxy(self))
```

Comment:
Is there a reason why we do not change how `Picker` keeps a reference to the scene? (See line https://github.com/dmsurti/mayavi/blob/fix-memory-leaks-bug/tvtk/pyface/picker.py#L242). I also wonder if `renwin` of Picker could be a WeakRef trait.